grails-app/utils/org/broadinstitute/orsp/utils/DulPdfParser.groovy

```diff
@@ -0,0 +1,228 @@
+package org.broadinstitute.orsp.utils
+
+import grails.converters.JSON
+import groovy.util.logging.Slf4j
+import org.apache.pdfbox.pdmodel.interactive.form.PDAcroForm
+import org.apache.pdfbox.pdmodel.interactive.form.PDField
+import org.broadinstitute.orsp.DataUseLetter
+import org.broadinstitute.orsp.dataUseLetter.DataUseLetterFields
+import org.springframework.util.StringUtils
+
+import java.text.SimpleDateFormat
+import java.util.regex.Pattern
+
+@Slf4j
+class DulPdfParser {
+    static PDAcroForm fillDulForm(DataUseLetter dul, PDAcroForm acroForm) {
+        Object dulInfoObj = JSON.parse(dul.dulInfo)
+        for (PDField field : acroForm.getFields()) {
+            String fieldName = field.getFullyQualifiedName()
+            switch (fieldName) {
+                case DataUseLetterFields.PROTOCOL_TITLE.abbreviation:
```

Comment:
I can see how some of this can be simplified a LOT, but the multi-layer checkbox ones are harder, I think. For the simple cases, this should work and cut this file down by at least half:

Define the grouped cases as a typed list:
```
        List<String> defaultValueFields = [
                DataUseLetterFields.PROTOCOL_TITLE.abbreviation,
                DataUseLetterFields.PROTOCOL_NUMBER.abbreviation,
                DataUseLetterFields.CONSENT_FORM_TITLE.abbreviation,
                DataUseLetterFields.PRINCIPAL_INVESTIGATOR.abbreviation,
                DataUseLetterFields.DATA_MANAGER_NAME.abbreviation,
                DataUseLetterFields.DATA_MANAGER_EMAIL.abbreviation,
                ...
        ]

        List<String> defaultDateValues = [
                DataUseLetterFields.START_DATE.abbreviation,
                DataUseLetterFields.END_DATE.abbreviation
        ]
```
Then, in the loop, loop over each one to set the value:
```
            defaultValueFields.each {
                if (fieldName.equals(it)) {
                    field.setValue(getDefaultValue(dulInfoObj[it] as String, dul.uid))
                }
            }
```
And repeat for each category of field value:
```
            defaultDateValues.each {
                if (fieldName.equals(it)) {
                    field.setValue(getDefaultDateValue(dulInfoObj[it] as String))
                }
            }
```
I think that would be simpler to read, but I don't have a great solution for the `(parseCheckBoxValue(dulInfoObj[DataUseLetterFields.DISEASE_RESTRICTED_OPTIONS.abbreviation][DataUseLetterFields.PARASITIC_DISEASE.abbreviation] as String))` cases

Reply:
It's a good idea! Please let me fix this in my next PR so we can make a deploy to dev environment. I can create a story for the coming sprint and then I'll assign it to my self.

Reply:
Sounds like a plan!